docs/tutorials/readout-error-mitigation-sampler.ipynb

```diff
@@ -0,0 +1,1414 @@
+    "a one-to-one relation to the classical bitstrings.\n",
+    "The $n$-qubit Hadamard operator on classical states can be viewed as a Fourier transform on the Boolean hypercube:\n",
+    "$$\n",
+    "H^{\\otimes n} =  \\frac{1}{\\sqrt{2^n}} \\sum_{x,y \\in {\\mathbb{Z}_2^n}} (-1)^{x \\cdot y} {|{y}\\rangle}{\\langle{x}|}.\n",
+    "$$\n",
+    "Consider a state ${|{s}\\rangle}$ corresponding to fixed bitstring $s$.\n",
+    "Applying $H^{\\otimes n}$, and using ${\\langle {x}|{s}\\rangle} = \\delta_{x,s}$,\n",
+    "we see that the Fourier transform of ${|{s}\\rangle}$ can be written as\n",
+    "$$\n",
+    "   H^{\\otimes n} {|{s}\\rangle} =  \\frac{1}{\\sqrt{2^n}} \\sum_{y \\in {\\mathbb{Z}_2^n}} (-1)^{s \\cdot y} {|{y}\\rangle}.\n",
+    "$$\n",
+    "\n",
+    "The Hadamard is its own inverse, that is,\n",
+    " $H^{\\otimes n} H^{\\otimes n} = (H H)^{\\otimes n} = I^{\\otimes n}$.\n",
+    "Thus, the inverse Fourier transform is the same operator, $H^{\\otimes n}$.\n",
+    "Explicitly, we have,\n",
+    "$$\n",
+    "  {|{s}\\rangle} =  H^{\\otimes n} H^{\\otimes n} {|{s}\\rangle}  =  H^{\\otimes n} \\frac{1}{\\sqrt{2^n}} \\sum_{y \\in {\\mathbb{Z}_2^n}} (-1)^{s \\cdot y} {|{y}\\rangle}.\n",
+    "$$"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d10fce93-3e08-4a27-b641-eda5e7b05ce0",
+   "metadata": {},
+   "source": [
+    "#### The hidden shift problem\n",
+    "\n",
+    "We consider a simple example of a _hidden shift problem_.\n",
+    "The problem is to identify a constant shift in the input to a function.\n",
+    "The function we consider is the dot product. It is the simplest member\n",
+    "of a large class of functions that admit a quantum speedup for the hidden shift\n",
+    "problem via techniques similar to those presented below.\n",
+    "\n",
+    "Let $x,y \\in {\\mathbb{Z}_2^m}$ be bitstrings of length $m$.\n",
+    "We define ${f}: {\\mathbb{Z}_2^m} \\times {\\mathbb{Z}_2^m} \\rightarrow \\{-1,1\\}$ by\n",
+    "$$\n",
+    "  {f}(x, y) = (-1)^{x \\cdot y}.\n",
+    "$$\n",
+    "  Let $a,b \\in {\\mathbb{Z}_2^m}$ be fixed bitstrings of length $m$.\n",
+    "  We furthermore define $g: {\\mathbb{Z}_2^m} \\times {\\mathbb{Z}_2^m} \\rightarrow \\{-1,1\\}$ by\n",
+    "$$\n",
+    "  g(x, y) = {f}(x+a, y+b) = (-1)^{(x+a) \\cdot (y+b)},\n",
+    "  $$\n",
+    "  where $a$ and $b$ are (hidden) parameters.\n",
+    "  We are given two black boxes, one implementing $f$, and the other $g$.\n",
+    "  We suppose that we know that they compute the functions defined above, except that we know\n",
+    "  neither $a$ nor $b$. The game is to determine the hidden bitstrings (shifts)\n",
+    "  $a$ and $b$ by making queries to $f$ and $g$. It's clear that if we play the game classically,\n",
+    "  we need $O(2m)$ queries to determine $a$ and $b$. For example, we can query $g$\n",
+    "  with all pairs of strings that, when combined, have exactly one element set to $1$.\n",
```

Comment:
Better:
For example, we can query $g$ with all pairs of strings such that one element of the pair is all zeros, and the other element has exactly one element set to $1$.